### PR TITLE
Add instructions for building docs

### PR DESCRIPTION
**Pull request by SeanKilleen**

### CODE REVIEW

**Comment by SeanKilleen** [comment]
@JeremySkinner I was attempting to build the docs using the instructions I added above since I saw no other guides.

When I run `make.bat html`, which I think should output the site, I see:

```
Running Sphinx v3.0.3

Configuration error:
There is a programmable error in your configuration file:

Traceback (most recent call last):
  File "c:\python38\lib\site-packages\sphinx\config.py", line 319, in eval_config_file
    execfile_(filename, namespace)
  File "c:\python38\lib\site-packages\sphinx\util\pycompat.py", line 81, in execfile_
    exec(code, _globals)
  File "C:\Users\SeanK\Repositories\FluentValidation\docs\conf.py", line 24, in <module>
    from recommonmark.parser import CommonMarkParser
ModuleNotFoundError: No module named 'recommonmark'
``` 

Do I need to install modules separately, or is there another correct way to build the docs?

Thanks! I figure once I get these in, I can help a little bit more to prep for 9.0

**Comment by JeremySkinner** [comment]
I’m not sure I’m afraid, Ive only ever run it on Linux, I’ve never tried to build it on windows. Have you worked through the Sphinx getting started documentation?

**Comment by JeremySkinner** [comment]
Think you probably just need to install it: https://recommonmark.readthedocs.io/en/latest/

**Comment by SeanKilleen** [comment]
@JeremySkinner ah, I see. I'll try on WSL for now. I looked at the Sphinx getting started docs but when I saw `make.bat` I figured it would take care of dependencies, etc. etc. 

Now that I know it's not handling that, I'll do it myself and see where it leads. 👍 Thanks!

**Comment by JeremySkinner** [comment]
Yeah I’m pretty sure the male command just builds the docs and doesn’t handle pip dependencies but can’t really remember (I’m not near a computer at the moment). It’s just the default one provided by the Sphinx project template

**Comment by SeanKilleen** [comment]
With `pip` dependencies installed to the best of my knowledge, the build still does not appear to work. I'll continue to troubleshoot and see if I can get the docs building on windows or WSL without 700+ warnings. :) 

Once we get this squared away, hopefully there will be a quick & easy path to contribute docs and see the preview locally.

**Comment by JeremySkinner** [comment]
To be honest most of the time running the full build locally is overkill and editing via the GitHub in-line editor is sufficient for text changes. Rarely do changes to the docs need any more than that.

**Comment by JeremySkinner** [comment]
Here are the steps I took to build the docs on a fresh clean WSL image:

Install python 3 and pip:

```
sudo apt install python3-pip
```

Install sphinx, sphinx-autobuild and recommonmark:

```
pip3 install -U sphinx
pip3 install -U sphinx-autobuild
pip3 install -U recommonmark
```

Make sure the pip packages directory is added to your PATH:

```
# Added to my .bashrc
PATH=$PATH:~/.local/lib/python3.8/site-packages
```

Run `make serve` in the docs directory.

**Comment by JeremySkinner** [comment]
There's also a breaking change between sphinx 2.x and 3.0  to do with compatibility with recommonmark, which causes the doc builds to fail if you install sphinx 3.x. I'll commit a change that should resolve that but just need to test it doesn't break RTD first.

**Comment by JeremySkinner** [comment]
By default readthedocs has a dependency on sphinx < 2, currently installing 1.8.5 which is several versions behind the default that gets installed with `pip install sphinx` (currently 3.0.4).

It may be worth considering either:
- Using a requirements file or setuptools to ensure that readthedocs installs a newer version of sphinx
- OR use a requirements file/setuptools to pin the local repo to the older version to match what's on RTD.

This would require some time to configure and experiment, which probably isn't worth it considering how rarely the docs need to be built locally. But it's something to consider.

**Comment by JeremySkinner** [comment]
So the install instructions should now be:

Install python3 & pip (on ubuntu can be done with `sudo apt install python3-pip`) 

In the docs directory run:

```
pip3 install -r requirements.txt
make serve
```

**Comment by SeanKilleen** [comment]
@JeremySkinner This is awesome! Thanks for the work on this.

Updated the docs with the text -- tried it on my Ubuntu WSL and it worked like a charm.

Should be good to go now. 👍

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -36,4 +36,16 @@ Make sure you can build the code. Familiarize yourself with the project workflow
...
-Tests must be provided for all pull requests that add or change functionality. 
+Tests must be provided for all pull requests that add or change functionality.
+
+## Building the Docs
+
+The docs are built separately from the source code. Building the docs requires Python 3 and pip. This is then used to install Sphinx and dependencies, which then enable `make` to build the site.
+
+For example, on Linux / within WSL:
+
+* `sudo apt install python3-pip`
```
Comment (JeremySkinner) [inline CONTRIBUTING.md:47]: Probably worth mentioning that the pip packages directory needs to be added to PATH.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -36,4 +36,16 @@ Make sure you can build the code. Familiarize yourself with the project workflow
...
+* `cd docs` to go to the docs directory
+* `pip3 install -r requirements.txt` to install the required packages for the docs
+* On WSL, you may need to exit and restart at this point.
+* `make html` to build the site or `make serve` to do auto watch.
```
Comment (JeremySkinner) [inline CONTRIBUTING.md:51]: "to watch for changes" rather than "to do auto watch"

**Comment by SeanKilleen** [comment]
@JeremySkinner I think this is good to go now. 👍

**Pull request merged**